brainiak/utils/simulator.py

```diff
@@ -0,0 +1,1058 @@
+    """
+
+    # Set up the input to the fast fourier transform
+    def fftIndgen(n):
+        a = list(range(0, int(n / 2 + 1)))
+        b = list(range(1, int(n / 2)))
+        b.reverse()
+        b = [-i for i in b]
+        return a + b
+
+    # Take in an array of fft values and determine the amplitude for those
+    # values
+    def Pk2(idxs):
+
+        # If all the indexes are zero then set the out put to zero
+        if np.all(idxs == 0):
+            return 0.0
+        return np.sqrt((np.sqrt(np.sum(idxs ** 2))) ** sigma)
+
+    noise = np.fft.fft2(np.random.normal(size=dimensions))
+    amplitude = np.zeros(dimensions)
+
+    for x, fft_x in enumerate(fftIndgen(dimensions[0])):
+        for y, fft_y in enumerate(fftIndgen(dimensions[1])):
+            for z, fft_z in enumerate(fftIndgen(dimensions[2])):
+                amplitude[x, y, z] = Pk2(np.array([fft_x, fft_y, fft_z]))
+
+    # The output
+    noise_spatial = np.fft.ifft2(noise * amplitude)
+
+    return noise_spatial.real
+
+
+def generate_noise(dimensions,
+                   stimfunction,
+                   tr_duration,
+                   noise_strength=[1]):
+    """ Generate the noise to be added to the signal
+
+    Parameters
+    ----------
+    dimensions : tuple
+        What is the shape of the volume to be generated
+
+    stimfunction :  Iterable, bool
+        When do the stimuli events occur
+
+    tr_duration : float
+        What is the duration, in seconds, of each TR?
+
+    noise_strength : list, float
+        Either a one element list or a list of length 3 which varies
+        the size of the noise for temporal noise, spatial noise and
+        system noise
+
+    Returns
+    ----------
+
+    multidimensional array, float
+        Generates the noise volume for these parameters
+
+
+    """
+
+    # Duplicate the noise strength if only one is supplied
+    if len(noise_strength) == 1:
+        noise_strength = noise_strength * 3
+
+    # What are the dimensions of the volume, including time
+    dimensions_tr = (dimensions[0],
+                     dimensions[1],
+                     dimensions[2],
+                     len(stimfunction))
+
+    # Generate the noise
+    noise_temporal = _generate_noise_temporal(stimfunction=stimfunction,
+                                              tr_duration=tr_duration,
+                                              ) * noise_strength[0]
+
+    noise_spatial = _generate_noise_spatial(dimensions=dimensions,
+                                            ) * noise_strength[1]
+
+    noise_system = _generate_noise_system(dimensions=dimensions_tr,
+                                          ) * noise_strength[2]
+
+    # Find the outer product and add white noise
+    noise = np.multiply.outer(noise_spatial, noise_temporal) + noise_system
+
+    return noise
+
+
+def mask_brain(volume,
+               mask_name="brainiak/utils/grey_matter_mask.npy"):
```

Comment:
Use `numpy.load(resource_stream(__name__, "grey_matter_mask.npy"))`. You are still hardcoding the name of the file, but not the path. Your current code only works when called from the root of the repository. When called from the examples directory or anywhere else, the hardcoded path is not valid:
`FileNotFoundError: [Errno 2] No such file or directory: 'brainiak/utils/grey_matter_mask.npy'`